typegate/src/runtimes/typegraph.ts

```diff
@@ -339,46 +376,41 @@ export class TypeGraphRuntime extends Runtime {
     }
 
     if (isEither(type) || isUnion(type)) {
-      const getVariants = (type: UnionNode | EitherNode) =>
-        isUnion(type) ? type.anyOf : type.oneOf;
-
-      const variants = getVariants(type);
-      const variantsAsObject = {
-        title: type.title,
-        type: "object",
-        properties: {},
-      } as ObjectNode;
-      let count = 0;
-      const objects = new Set<[string, number]>();
-      const remaining = new Set<[string, number]>();
-      for (let i = 0; i < variants.length; i++) {
-        const idx = variants[i];
-        const node = this.tg.types[idx];
-        if (isObject(node)) {
-          for (const [field, idx] of Object.entries(node.properties)) {
-            objects.add([field, idx]);
-          }
-        } else {
-          // name for scalars and nested union/either
-          const field = `${type.type}_${count++}`;
-          remaining.add([field, idx]);
-        }
-      }
+      const variants = isUnion(type) ? type.anyOf : type.oneOf;
+      if (asInput) {
+        const titles = new Set<string>(
+          variants.map((idx) => this.tg.types[idx].title),
+        );
+        const description = `${type.type} type\n${
+          Array.from(titles).join(", ")
+        }`;
 
-      for (const [field, idx] of objects) {
-        variantsAsObject.properties[field] = idx;
-      }
-      for (const [field, idx] of remaining) {
-        variantsAsObject.properties[field] = idx;
+        return {
```

Comment:
Can you add comment explain why we are making this trick for all cases?

Reply:
When a either/union node is on the input, we make it so that it is seen as a custom scalar when introspected, the description will inform about the variants. 
When it is on the output, it is seen as a native `UNION`, but since `UNION` can only support objects, we have to map the scalar variants to a custom object, for example `int -> { title: "_Integer", fields: { integer: typeIdx }}`.

Reply:
The trickiest part is handling scalars on the output, when we are sure that the processed type has union/either, we have to generate the custom object types beforehand (`hasUnion` flag), this way when an object is processed with `formatType`, we don't have to worry about the missing types anymore since we cannot refer to types that does not exist in the schema list.